content/spin/http-trigger.md

```diff
@@ -92,45 +119,188 @@ will be handled by `component-1`:
+Every HTTP application automatically has a special route always configured at `/.well-known/spin/health`, which
 returns `OK 200` when the Spin instance is healthy.
 
-Once Spin selects a component to handle an incoming request based on the route
-configuration, it will instantiate and execute that component based on its
-defined _HTTP executor_, and the next sections explore the two ways of building
-HTTP components based on the two available executors.
+## HTTP Components
+
+> Spin has two ways of running HTTP components, depending on language support for the evolving WebAssembly component standards.  This section describes the default way, which is currently used by Rust, JavaScript/TypeScript, Python, and TinyGo components.  For other languages, see [HTTP Components using Wagi](#http-components-using-wagi) below.
+
+By default, Spin runs components using the [WebAssembly component model](https://github.com/WebAssembly/component-model).  In this model, the Wasm module exports a well-known function that Spin calls to handle the HTTP request.
+
+### The Request Handler
+
+The exact signature of the HTTP handler, and how a function is identified to be exported as the handler, will depend on your language.
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+In Rust, the handler is identified by the `#[spin_sdk::http_component]` attribute.  It takes a `spin_sdk::http::Request`, and returns a `spin_sdk::http::Response` (or error).  These types are instantiations of the standard `http::Request` and `http::Response` types and behave exactly like them.
+
+<!-- @nocpy -->
+
+```rust
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+};
+
+/// A simple Spin HTTP component.
+#[http_component]
+fn handle_hello_rust(req: Request) -> Result<Response> {
+    Ok(http::Response::builder()
+        .status(200)
+        .header("foo", "bar")
+        .body(Some("Hello, Fermyon".into()))?)
+}
+```
+
+{{ blockEnd }}
+
+{{ startTab "TypeScript"}}
+
+In JavaScript or TypeScript, the handler is identified by name.  It must be called `handleRequest`.  The way you declare it is slightly different between the two languages.
+
+In **JavaScript**, `handleRequest` is declared as `export async function`.  It takes a JsvaScript object representing the request, and returns a response object.  The fields of these objects are exactly the same as in TypeScript.
+
+<!-- @nocpy -->
+
+```javascript
+const encoder = new TextEncoder()
+
+export async function handleRequest(request) {
+    return {
+        status: 200,
+        headers: { "foo": "bar" },
+        body: "Hello from JS-SDK"
+    }
+}
+```
+
+In **TypeScript**, `handleRequest` is declared as an `export const` of the `HandleRequest` function type - that is, a function literal rather than a function declaration.  It takes a `HttpRequest` object, and returns a `HttpResponse` object, both defined in the `@fermyon/spin-sdk` package.
```

Comment:
```suggestion
In **TypeScript**, `handleRequest` is declared as an `export const` of the `HandleRequest` function type - that is, a function literal rather than a function declaration.  It takes a `HttpRequest` object, and returns a `HttpResponse` object, both defined in the `@fermyon/spin-sdk` package:
```